.github/workflows/foreman.yml

```
@@ -0,0 +1,65 @@
+---
+name: Foreman
+
+on:
+  - pull_request
+
+env:
+  RAILS_ENV: test
+  DATABASE_URL: postgresql://postgres:password@localhost/test
+  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+  BUNDLE_WITHOUT: "console:development:journald"
+
+jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: theforeman/gha-matrix-builder@v0
+  test:
+    name: "develop with Ruby ${{ matrix.ruby }} and Node ${{ matrix.node }} on PostgreSQL 12"
+    runs-on: ubuntu-latest
+    needs: setup_matrix
+    services:
+      postgres:
+        image: 'postgres:${{ matrix.postgresql }}'
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        env:
+          POSTGRES_PASSWORD: password
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.setup_matrix.outputs.matrix) }}
+    steps:
+      - run: sudo apt-get update
+      - run: sudo apt-get install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev libvirt-dev
+      - name: "Check out Foreman develop"
```

```suggestion
      - name: "Check out Foreman"
```

This will check out whatever branch this PR is against, not `develop` :)